Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -312,12 +312,17 @@ void SoftwareGLContext::gl_end()
 
     // Set up normals transform by taking the upper left 3x3 elements from the model view matrix
     // See section 2.11.3 of the OpenGL 1.5 spec
-    auto const& mv_elements = m_model_view_matrix.elements();
-    auto normal_transform = FloatMatrix3x3(
-        mv_elements[0][0], mv_elements[0][1], mv_elements[0][2],
-        mv_elements[1][0], mv_elements[1][1], mv_elements[1][2],
-        mv_elements[2][0], mv_elements[2][1], mv_elements[2][2]);
-    normal_transform = normal_transform.inverse();
+    auto const& inverse_of_model_view_matrix = m_model_view_matrix.inverse();
+    FloatMatrix3x3 normal_transform;
+    if (inverse_of_model_view_matrix.is_error()) {
```

Comment:
Is there a case for checking for this? This feels like a case of "Well, the programmer messed up!" instead of us just letting it through and returning a GL Error code (similar to how we don't check params for `NULL` and directly do a deref on them)

Reply:
GLTron sets an invalid matrix a couple of seconds into the gameplay, causing the old code to crash. The new behavior is based on what TinyGL does (actually, that sets the unmodified model view matrix as the normal transformation matrix). I don't think this warrants returning with an exit code, since we are only missing a proper normal transformation matrix.

Reply:
How about wie disable floating point exceptions and just return Inf.

Reply:
A matrix full of Inf would still not be the inverse, right? At least this way we know that something is wrong.

Reply:
I'll have to check against NVIDIA's implementation. But I think returning Infs/NaNs is completely expected.